proof/taproot.go

```diff
@@ -579,3 +580,62 @@ func (p TaprootProof) DeriveByTapscriptProof() (*btcec.PublicKey, error) {
 	}
 	return p.TapscriptProof.DeriveTaprootKeys(p.InternalKey)
 }
+
+// AddExclusionProofs adds exclusion proofs to the base proof for each P2TR
+// output in the given PSBT that isn't an anchor output itself.
+func AddExclusionProofs(baseProof *BaseProofParams, packet *psbt.Packet,
```

Comment:
I think this is in the BIP master tracking issue, but we should update the BIP where relevant to mention the proper additional checks needed.

Reply:
I think this is already covered properly by the `taproot_exclusion_proofs` [here](https://github.com/Roasbeef/bips/blob/bip-taro/bip-taro-proof-file.mediawiki#specification). The spec was ahead of the implementation here because we didn't get P2TR change outputs in previous versions of `lnd`, so we didn't implement it at the time we added the proof logic.